numpy/_core/src/umath/ufunc_object.c

```diff
@@ -115,10 +115,11 @@ static PyObject *
 prepare_input_arguments_for_outer(PyObject *args, PyUFuncObject *ufunc);
 
 static int
-resolve_descriptors(int nop,
+resolve_descriptors(int nop, int nin,
```

Comment:
Hmm, I think I'd still remove this extra addition of `nin`, since you only need it for the private case: just use `ufunc->nin` there.  (Of course, no choice for `inputs_tup`!)